1050 add audit log support

jeaaustx opened the pull request:
Pull in all of the commits needed to support audit logging in bmcweb. These are all previously merged in 1030/1040.

This introduces a new bmcweb meson option, audit-events, which is defaulted to disabled. A later commit to openbmc/openbmc will enable audit logging on 1050. 

When audit-events is enabled all bmcweb PATCH, POST, PUT, and DELETE events are logged using the Linux kernel auditd subsystem. Additionally, login events coming through bmcweb are also recorded.

The body of the events is recorded as well except in the following cases:
- /redfish/v1/AccountService/Accounts PATCH/POST events - body is not recorded as it may contain clear text password
- /ibm/v1 PATCH/POST/PUT events - body is not recorded as it contains HMC config file binary data

Events are recorded in /var/log/audit/. User type dreport will gather these log files. See: [PR #100 phosphor-debug-collector](https://github.com/ibm-openbmc/phosphor-debug-collector/pull/100)

Upstream: [Discussion has begun for upstream work.](https://gerrit.openbmc.org/c/openbmc/bmcweb/+/55532) I will be addressing the community concerns brought up there as well as updating with the current state of this implementation.

Tested: Enabled auditing then initiated a variety of Redfish events using curl. Confirmed the events were recorded and confirmed the data recorded was accurate. Confirmed password was not included in audit data recorded.

gtmills commented:
@jeaaustx Can you expand the PR msg a bit more? Do you have an openbmc/openbmc commit ? How did you test? 
I added a sentence to the PR msg (feel free to edit)

jeaaustx commented:
Thank you Gunnar for the suggestions on improving the PR message, definitely all good information I should have had in there in the first place. I've added more detail to it and also adjusted the title to better reflect what this drop actually does.
